### PR TITLE
chore(deps): use shared eslint configuration

### PR DESCRIPTION
**Pull request by steabert**

### CODE REVIEW

**Review by lekoaf — APPROVED** [review]
LGTM

File: lib/Settings.tsx [hunk 1]
```diff
@@ -77,33 +80,51 @@ export const Settings: React.FC<SettingsProps> = ({
...
+    [onFormat],
+  )
+
+  const changeResolution = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      if (e.target.value === '') {
+        onVapix('resolution', '')
+        return
+      }
+      onVapix('resolution', e.target.value)
```
Comment (Tigge) [inline lib/Settings.tsx:96]: The if statement here seems a bit unnecessary?

File: lib/StillImage.tsx [hunk 2]
```diff
@@ -51,20 +51,28 @@ export const StillImage: React.FC<StillImageProps> = ({
...
         imgRef.current.src = ''
         unsetLoaded()
       }
     }
-  }, [play, src])
+  }, [play, src, unsetLoaded])
+
+  // keep a stable reference to the external onPlaying callback
+  const __onPlayingRef = useRef(onPlaying)
+  __onPlayingRef.current = onPlaying
```
Comment (Tigge) [inline lib/StillImage.tsx:66]: This change seems a bit weird... why? Just ignore the lint rule for dep array if that is the problem.

File: lib/WsRtspCanvas.tsx [hunk 3]
```diff
@@ -72,40 +72,51 @@ export const WsRtspCanvas: React.FC<WsRtspCanvasProps> = ({
...
         setPipeline(null)
         setFetching(false)
         debugLog('canvas cleared')
       }
     }
   }, [ws, rtsp])
 
+  // keep a stable reference to the external onPlaying callback
+  const __onPlayingRef = useRef(onPlaying)
+  __onPlayingRef.current = onPlaying
```
Comment (Tigge) [inline lib/WsRtspCanvas.tsx:95]: See other comment

File: lib/WsRtspCanvas.tsx [hunk 4]
```diff
@@ -72,40 +72,51 @@ export const WsRtspCanvas: React.FC<WsRtspCanvasProps> = ({
...
-      })
+      pipeline.ready
+        .then(() => {
+          debugLog('fetch')
+          pipeline.rtsp.play()
+          setFetching(true)
+        })
+        .catch((err) => {
+          console.error(err)
+        })
```
Comment (Tigge) [inline lib/WsRtspCanvas.tsx:107]: .catch(console.error) ?

File: lib/WsRtspVideo.tsx [hunk 5]
```diff
@@ -89,6 +89,10 @@ export const WsRtspVideo: React.FC<WsRtspVideoProps> = ({
   )
   const [fetching, setFetching] = useState(false)
 
+  // keep a stable reference to the external onPlaying callback
+  const __onPlayingRef = useRef(onPlaying)
+  __onPlayingRef.current = onPlaying
```
Comment (Tigge) [inline lib/WsRtspVideo.tsx:94]: See other comment

**Review by Tigge — CHANGES_REQUESTED** [review]

File: lib/StillImage.tsx (hunk 2 above)
Comment (steabert) [reply lib/StillImage.tsx:66]: I want to use the latest updated `onPlaying` callback, but not re-run the effect, so I'm keeping a reference to it to be used inside the effect. If it's ignored from the dep array, the callback itself will not be updated.

File: lib/Settings.tsx (hunk 1 above)
Comment (steabert) [reply lib/Settings.tsx:96]: Indeed, would be interesting to see the history of this.

Comment (lekoaf) [reply lib/Settings.tsx:96]: `onVapix('resolution', '')`

Results in...

```
if (value === '') {
  delete p[key]
  return { ...p }
}
```

Comment (lekoaf) [reply lib/Settings.tsx:96]: I may or may not be the culprit of this less than ideal solution. :smile:

Comment (steabert) [reply lib/Settings.tsx]: Fixed now.

**Pull request merged**